apps/opencs/model/world/commands.cpp

```diff
@@ -15,6 +16,173 @@
 #include "nestedtablewrapper.hpp"
 #include "pathgrid.hpp"
 
+CSMWorld::TouchCommand::TouchCommand(IdTable& table, const std::string& id, QUndoCommand* parent)
+    : QUndoCommand(parent)
+    , mTable(table)
+    , mId(id)
+    , mOld(nullptr)
+    , mChanged(false)
+{
+    setText(("Touch " + mId).c_str());
+    mOld.reset(mTable.getRecord(mId).clone());
+}
+
+void CSMWorld::TouchCommand::redo()
+{
+    mChanged = mTable.touchRecord(mId);
+}
+
+void CSMWorld::TouchCommand::undo()
+{
+    if (mChanged)
+    {
+        mTable.setRecord(mId, *mOld);
+        mChanged = false;
+    }
+}
+
+CSMWorld::ImportLandTexturesCommand::ImportLandTexturesCommand(IdTable& landTable,
+    IdTable& ltexTable, QUndoCommand* parent)
+    : QUndoCommand(parent)
+    , mLands(landTable)
+    , mLtexs(ltexTable)
+    , mOldState(0)
+{
+    setText("Import land textures");
+}
+
+void CSMWorld::ImportLandTexturesCommand::redo()
+{
+    int pluginColumn = mLands.findColumnIndex(Columns::ColumnId_PluginIndex);
+    int oldPlugin = mLands.data(mLands.getModelIndex(getOriginId(), pluginColumn)).toInt();
+
+    // Original data
+    int textureColumn = mLands.findColumnIndex(Columns::ColumnId_LandTexturesIndex);
+    mOld = mLands.data(mLands.getModelIndex(getOriginId(), textureColumn)).toByteArray();
```

Comment:
Just noticed that you are using ByteArray all over the place. Is there a specific reason not to simply use std::vector instead? We generally prefer C++ STL containers over Qt containers, unless there is a very good reason to do otherwise.

Reply:
As far as I know, you can't store an STL container in a QVariant.

Reply:
There is probably a way to do that, but that is actually a good enough reason to not use STL containers.

Reply:
I've been looking into it a bit, and it is indeed possible. I had previously come to the mistaken conclusion that QVariant was very limited with what could be passed. I'll make an attempt to use an STL container, but I can at least get rid of the need for reinterpret_cast.